### ISSUE TITLE
libvirt - Failed to open file '/lib/modules/x.x.x/modules.alias'

### ISSUE THREAD

**Issue by ZaneA** [NixOS/nixpkgs#285929]
### Describe the bug
When `<hostdev managed="yes">` devices are present, libvirt 10 attempts to open "/lib/modules/x.x.x/modules.alias" which doesn't exist on NixOS.

From https://libvirt.org/news.html#v10-0-0-2024-01-15 :

*qemu: automatic selection/binding of VFIO variant drivers*

*When a device is assigned to a guest using VFIO with <hostdev managed='yes'>, libvirt will now search the running kernel's modules.alias file for the most specific match to that device for a VFIO driver, and bind that driver to the device rather than vfio-pci. A specific driver can also be forced, using the <driver model='plugh'/> attribute.*

### Steps To Reproduce
1. Have a VM using VFIO, specifically one or more devices with `<hostdev managed="yes">`.
2. Launch the VM.

### Expected behavior
The VM starts as usual.

### Additional context
It looks like this just needs [another substituteInPlace](https://github.com/NixOS/nixpkgs/blob/e718d5841ae8cae2ef9b6917477b6459318f6a97/pkgs/development/libraries/libvirt/default.nix#L250) such as:

```nix
substituteInPlace src/util/virpci.c \
  --replace '/lib/modules' '/run/current-system/kernel-modules/lib/modules'
```

(**edit**: yeah this fixes it for me locally)

### Notify maintainers
@fpletz @globin @lovesegfault

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.2, NixOS, 24.05 (Uakari), 24.05pre578631.b8b232ae7b8b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(zanea): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by fpletz** [comment]
Thanks for reporting this. The fix looks acceptable. Do you want to open a PR? :)

**Comment by brenix** [comment]
I've have the same issue and substituting the string works for me as well. I've gone ahead and opened a [PR](https://github.com/NixOS/nixpkgs/pull/286112) with these changes

**Comment by ZaneA** [comment]
Appreciated! Now that I'm familiar with rebuilding NixOS using a local nixpkgs hopefully I can jump directly to the PR in the future, thanks all :)

**Comment by linyinfeng** [comment]
The issue can also be reproduced with `virsh nodedev-detach`, so we don't need to create a VM for testing.

```console
# virsh nodedev-detach pci_0000_00_02_1 
error: Failed to detach device pci_0000_00_02_1
error: Failed to open file '/lib/modules/6.6.14/modules.alias': No such file or directory
```

**Comment by fpletz** [comment]
Fixed by #287544.
